src/server/plugins/internal/firestore.ts

```diff
@@ -40,6 +47,11 @@ export interface Options {
       onUpdate?: TransactionHandler
       onRemove?: TransactionHandler
     }
+    consents: {
+      onCreate?: ConsentHandler
```

Comment:
all optional properties smell bad -> later there is a need to always check is it present before a use

Reply:
hi pawel - can you raise an issue for this - i just made style changes here and for the sake of separation of concerns I don't want to modify code not related to this ticket/PR

Reply:
@stevenwjy is there a reason we made the oncreate and onupdate optional?

Reply:
Hmm, actually there's no specific reason... Since it is a plugin, I just thought to make it as customizable as possible for people who use the plugin, which means they could decide to not handle the create operation etc. This is like the implementation of some button widgets in the `pisp-demo-server-app`, typically the `onTap` parameter is optional, in case the button shouldn't do anything when people tap it (probably @kkzeng can clarify). However, considering that this is currently an internal plugin, I shouldn't be an issue to change it into a required option.

Reply:
@jgeewax  any thoughts?

Reply:
I'm not 100% sure about this. Optional isn't necessarily bad, particularly when we actively just want "the default behavior". In this case, this same library is the only one accessing this property. And, I'm no TS expert, but I thought as of TS 3.7 you had the safe navigation/calling operator?

I'm talking about the `?.` operator:
  - `foo?.bar` will be `null` if `foo` is `null`
  - `foo.bar?.()` will call `foo.bar()` if it's not null, and do nothing if it's `null`

```typescript
if (change.type === 'added') {
  consentHandlers.onCreate?.(server, {
    id: change.doc.id,
    ...change.doc.data(),
  })
}
```